database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -76,13 +71,19 @@ public boolean createTable(Class<? extends BaseEntity> entityClass) {
 
         LOG.info("Executing SQL: {}", createTableSQL);
 
-        try (Connection connection = DriverManager.getConnection(databaseURL, dbUserName, dbPassword);
-             Statement statement = connection.createStatement()) {
+        Connection connection = null;
+        Statement statement = null;
+
+        try {
+            connection = connectionPool.getConnection();
+            statement = connection.createStatement();
             statement.executeUpdate(createTableSQL.toString());
-            return checkTableExists(databaseName, tableName);
+            return checkTableExists(databaseName, tableName, connection);
         } catch (SQLException e) {
             LOG.error("Unable to create new table: {}", e.getMessage());
             throw new CreationDatabaseException(UNABLE_CREATE_TABLE + ": " + e.getMessage());
+        } finally {
+            closeConnectionStatementResources(connection, statement);
```

Comment:
You don't need to manually close statements and result sets, you can use try-with-resources for them. You just need to add finally block where you're releasing the connection.
Something like:

```
Connection connection = connectionPool.getConnection();
try(Statement statement = connection.createStatement()) {
    //do something
} finally {
    connectionPool.releaseConnection(connection);
}
```
In this case, Java executes close methods for all resources created in try(...), and after that executes your custom finally block. Check the official Oracle documentation: [The try-with-resources Statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html#:~:text=The%20resource%20java.,like%20an%20ordinary%20try%20statement.). Please fix for all methods.